### PR TITLE
feat: add support for different compressions and levels

### PR DESCRIPTION
**Pull request by vetsin**
Downstream project would like to supply compresslevel per https://docs.python.org/3/library/tarfile.html

Also decided why not support the other modes?

### CODE REVIEW

File: securetar/__init__.py [hunk 1]
```diff
@@ -56,12 +67,16 @@ def __init__(
 
     def __enter__(self) -> tarfile.TarFile:
         """Start context manager tarfile."""
+        extra_args = {}
+        if self._tar_mode in ('w:gz', 'r:gz', 'w:bz2', 'r:bz2', 'x:gz', 'x:bz2'):
+            extra_args['compresslevel'] = self._compresslevel
```
Comment (pvizeli) [inline securetar/__init__.py:72]: Because of encryption, we need open it as an stream `:` will not work at all.

File: securetar/__init__.py [hunk 2]
```diff
@@ -44,7 +45,17 @@ def __init__(
 
         # Tarfile options
         self._tar: Optional[tarfile.TarFile] = None
-        self._tar_mode: str = f"{mode}|gz" if gzip else f"{mode}|"
+        if isinstance(gzip, bool):
```
Comment (pvizeli) [inline securetar/__init__.py:48]: let's rewrite with using Enum for compression and not doing it like this

**Review by pvizeli — COMMENTED** [review]
It need for each case a own test case with and without encryption to see if the streaming part works

**Comment by agners** [comment]
Closing since it went stale and needs rebase at this point.

**Pull request closed without merging**